pytorch_lightning/utilities/seed.py

```diff
@@ -48,13 +48,13 @@ def seed_everything(seed: Optional[int] = None, workers: bool = False) -> int:
     max_seed_value = np.iinfo(np.uint32).max
     min_seed_value = np.iinfo(np.uint32).min
 
-    try:
-        if seed is None:
-            seed = os.environ.get("PL_GLOBAL_SEED")
-        seed = int(seed)
-    except (TypeError, ValueError):
-        seed = _select_seed_randomly(min_seed_value, max_seed_value)
-        rank_zero_warn(f"No correct seed found, seed set to {seed}")
+    if seed is None:
+        global_seed = os.environ.get("PL_GLOBAL_SEED")
+        if isinstance(global_seed, str) and all(char.isdigit() for char in global_seed):
+            seed = int(global_seed)
```

Comment:
I believe the changes here are not equivalent to the previous code. 
If this condition does not hold, we select no seed while previously we would handle it in the except block. 
Anything standing against keeping it? 

Generally I would suggest not letting mypy force changes of our code flow.

Reply:
I've also been suspecting the code behaviour might be a bit changed :/ . The code can be adjusted to ensure the same functionality, however, it's gonna be more complex than the original solution. Maybe mypy can be asked not to check the `seed = int(seed)` statement?